### ISSUE TITLE
Feedback on communications

### ISSUE THREAD

**Issue by tracykteal** [carpentries/conversations#1]
Software and Data Carpentry have at their core a collaboration-driven ethos, and communication is key to that collaboration. We’re reaffirming our commitment to open and transparent communication, because we know we can do better! We want to give community members opportunities to talk to each other, staff and Steering Committees, to get updates on efforts and activities and to generate ideas and participate in discussions.

So, first, we want to hear from you!

What ideas do you have about communication? What do you want to hear from us? What channels do you like to use for communication? Do you like email lists or forums that include every topic, or ones on particular questions, domains or regions? What do you like about communication now? What don’t you like?

We’re going to be working on communication channels and strategies to promote and support these ideas, and continue to make the Carpentries a community that you are excited to be a part of, so please let us know what you think!

Please respond here in this issue, or start a new issue in this repository (we're considering this our suggestion box) if you have particular topics. Thanks for your feedback!

**Comment by ErinBecker** [comment]
As an organization, the Carpentries are committed to making our communications more transparent and doing so in a community-driven way. The Carpentry community runs on volunteer effort. The majority of our organization’s activities are carried out by a large, geographically
discontinuous group of individuals with varying interests and levels of time to commit to Carpentry work. As an organization we also have a “collaboration-driven ethos”. To be true to this ethos, and effective in our mission, our community needs to be able to communicate effectively about both aspirations and ongoing efforts. Effective communication allows us to learn from each other, identify critical issues, recover quickly from mistakes, evaluate ideas and commitments and make strategic decisions.

As a community, we communicate in many ways and for different purposes.

- Community members take initiative to coordinate activities.  
- Staff and committee members seek community input.  
- Staff and committee members report actions and deliver products to the community.  

We know that we need effective ways for:

- Community members to propose ideas for new work or directions for ongoing work.  
- Community members to organize work efforts around a particular issue.  
- Community members to stay up-to-date on work going on in the community, including work done by staff members,  
- Steering Committees and subcommittees and unofficial groups of community members.
- Staff to jointly decide on priorities, form productive collaborations and keep up-to-date on progress of projects.  

We also know that there may be other communication needs we have as an organization that we haven't yet considered. We invite anyone who has experience in communications, in building open communities or who simply has thoughts about these issues to contribute as we
work to develop an thoughtful, efficient and transparent communications strategy.

We envision this repository as a first step in developing this strategy. We propose that this repository become the canonical channel for communicating new ideas, suggestions, and progress updates between Carpentry community members and staff.

To take part in the conversation about developing communication strategy - please respond to this issue.

To start another conversation about a suggestion you have, ask a question about ongoing work, or give feedback about an issue or obstacle you face in the community, please start a new Issue.

As we work to develop a communications strategy, Carpentry staff will actively monitor this repo and follow-up on issues. All decisions about communications will "live" on this repo. If the community-driven process leads to a decision to use a mechanism other than GitHub for facilitating communications, this repo will be retired and communications migrated to a permanent canonical channel.

**Comment by wking** [comment]
> * Steering Committees and subcommittees and unofficial groups of community members.

The rest of the entries in that list are things that increased communication can help with, following the pattern “${actor} to ${task}…”.  This one is just a list of actors.  Maybe it's trying to be something like:

> * Steering Committee members to stay up-to-date with work by subcommittees, task forces, and unofficial groups of community members.

**Comment by ErinBecker** [comment]
Thanks @wking - it was a mis-placed line break. Now fixed.

**Comment by wking** [comment]
I'm fine with a discussion-only repo as a place to organize initial discussion or report on external discussions.  But I think you lose something if you keep *all* discussion here.  For example, I'd rather see the steering committee and subcommittees keeping each-other up to date by publishing their meeting minutes (e.g. [here][1], [here][2], …).  And subcommittee/task-force proposals, by-law changes, etc. are good fits for PRs (swcarpentry/board#148, swcarpentry/board#149).  Keeping them in issues (vs. PRs) makes it harder to track changes to the proposal as it evolves.

However, there is a clear benefit to having a pan-carpentry repo to host pan-carpentry ideas and discussion (and potentially pan-carpentry subcommittees like the policy subcommittee?).  I'd recommend we fit this repository into the framework I was sketching out [here][3], by directing pan-carpentry discussion and updates here (so where I'd said “this repo”, mostly replace it with “carpentries/conversations”).  But once the idea/suggestion gets to the point where a useful PR can be made, discussion should move to the repository where the change will land (e.g. swcarpentry/board for a new SWC subcommittee) and the carpentries/conversations would get a comment (“discussion moved to swcarpentry/board#whatever) and be closed if/when that PR landed.

For ideas that spring to life as a PR (e.g. swcarpentry/board#148), we may want to publish issues here with just a pointer to the external PR.  That could get noisy if it happens for *every* carpentry PR, and I'm not sure what a good rule of thumb would be to balance discovering pending changes vs. flooding with minor details ;).  Are we comfortable just using this repository for a catch-all for changes where contributors can't figure out the more appropriate target repository (possibly because a more appropriate target repository doesn't exist)?

[1]: https://github.com/swcarpentry/board/tree/73920cb2369b502d7780e4957943a809401925b3/minutes
[2]: https://github.com/swcarpentry/board/tree/73920cb2369b502d7780e4957943a809401925b3/subcommittees/mentoring/minutes
[3]: https://github.com/swcarpentry/board/issues/140#issuecomment-265088702

**Comment by wking** [comment]
Also, we need to be as clear as possible in scoping this repository in relation to the existing [carpentries/bridge][1].  It's currently not clear to me if they are actually addressing distinct ideas.

[1]: https://github.com/carpentries/bridge

**Comment by tracykteal** [comment]
Thanks @wking. We ultimately probably won't keep all communication here. This is a seed repo to collect ideas so we can have a more comprehensive communications strategy, so things don't get distributed and lost, as you've noted. Thanks for all the ideas so far, and we look forward to hearing ideas and perspectives from you and other community members as well!

**Comment by erjank** [comment]
> What channels do you like to use for communication?

I like receiving emails to get announcements and as a place for discussion with a few (1-4) individuals at a time. Past a few individuals, emails get messy, and solutions like moving discussions to github repositories seems preferable. 

> What do you want to hear from us? 

I want to hear _when_ things are happening. E.g., When there's been a big change to a core repository, or when people are organizing events at a conference, or when there's a deadline that would benefit from the community's attention, or when something blew up and we need to fix it.

> Do you like email lists or forums that include every topic, or ones on particular questions, domains or regions? 

I like email lists, except when it gets hard to follow a discussion.
I don't find myself frequenting forums very much.

> What do you like about communication now? 

I like that there exists a strategy to move in-depth discussion with many people to repositories (whether or not this is the optimal strategy). I think it goes a long way towards saying "Let's be respectful of inboxes", which is consistent with a culture of being nice. 

> What don’t you like?

Even with the existing efforts to respect inboxes, there is far more software-carpentry content that comes into my inbox than I am able to parse. Beyond that, there is sufficiently voluminous software-carpentry content online that it is not always easy to find what I'm looking for, whether it is a particular exchange in a conversation, a repository, or information on instructor training.


I'll add that I think this is an important topic to solicit feedback on, but it felt big enough to me that I didn't know where to start replying, or if my feedback was within the scope of what you're soliciting. Maybe others feel the same? I think I expected to see a lot more discussion here, given how much discussion there is on the [Discuss] email list.

**Comment by rgaiacs** [comment]
> What ideas do you have about communication?

Did anyone do an assessment research on the communication channels/tools available for us to use?

I believe that the important points are

1. how easy is for newcomers find the channel they are interested to keep a eye on? For example, using a single mailing list server has the advantage of the [list](http://lists.software-carpentry.org/) and the same is true for [Discourse](http://discuss.datacarpentry.org/categories) but I think GitHub and Google Groups a little more difficult since the search features brings more non-related channels.

2. how easy is to search for old information? For example, from time to time I want/need to find an old conversation and a single mailing list server or Discourse help a lot since I only need to search in one place (for the mailing list if Google has indexed that domain) but in GitHub I need to search in more than one organisation and each organisation has more than one repository (maybe I need classes on how to use GitHub search feature) and Google Groups brings a lot of non-related content.

3. how easy is to mute some threads where an flame war started? For example, mailing list is the worse tool for it because you need to be a Jedi on your email client but Discourse, GitHub and Google Groups (I think) offer easy options to mute one thread and not the hole channel.

4. option to integrate with peoples actual workflow. We don't want to say to people to install program X and create account on service Y giving access to account Z to be volunteer for us. We want to say, if you use X or Y or Z you can be part of our group.

> What do you want to hear from us?

- Discussions,
- Reflections,
- Cases, and
- Opportunities.

> What channels do you like to use for communication?

I like to keep my communication on my email box just because at the moment is the only option for me to keep most of the communication on the same place where I can access. I think instant message quite distracting and I try to avoid using then some times.

> Do you like email lists or forums that include every topic, or ones on particular questions, domains or regions?

I'm fine with email lists and forums.

> What do you like about communication now?

Is easy to say what I don't like.

> What don’t you like?

1. Slack.

   The only messages that I get on it is Software Carpentry blog posts announcements. If now one is using we should drop it.

2. Missing some good conversation on Data Carpentry.

   I didn't subscribed to all Data Carpentry GitHub repositories and I think that I should subscribe to some but I don't know which ones.

**Comment by leobrowning92** [comment]
> What ideas do you have about communication?

I think that for an organization like this, that has a wide range of levels of involvement from the community it is vital to be able to have some personal control over when and how much communications one is engaging with.
> What channels do you like to use for communication? 

I like to use email for directed conversation. chains are unwieldy and clutter my inbox
I like slack for group discussions and announcements, as it allows very easy selection on the user and host end as to what content to access.
I think github issues should be restricted to issues with repositories. it is too fragmented, especially in terms of cross repository search to function as a forum.
> What do you like about communication now? 

I like that there is a genuine effort to have open communication channels that make it possible for community members to be heard.
> What don’t you like?

Too many fragmented channels of communication (mailing list, github issues, slack?)

**Comment by serahrono** [comment]
It has been two years since this discussion started, and several changes have happened in that time, key among them being that [Library Carpentry became an official Lesson Program of The Carpentries in November 2018](https://carpentries.org/blog/2018/11/welcoming-library-carpentry/). 

We are currently working on updating The Carpentries communications strategy, and as such, we would like to revisit this exchange and are keen to hear from you on a number of things:

- what’s currently working, and what’s not, about
  *  the ways we communicate with you as members our community, and 
  * the ways we enable you to interact with others in the community?

-  what aspects of our communiqué do you still have questions about or need clarity on, be it information in a resource we've shared with you, or a page on our website, etc?

- do you have any suggestions on how we can 
  * communicate more effectively with you going forwards and/or
  * facilitate better interactions between you and other members of The Carpentries community?

Your feedback on any of the questions above really matters to us, and we will inform our comms strategy drafting process and implementation later on.

[Submissions via email](mailto:serah@carpentries.org) are also welcome.

**Comment by ctb** [comment]
as a professor, here are my thoughts :)

periodic e-mails are great for general notifications!

twitter and blogging remain my main officially supported mechanisms for keeping up with the Carpentries goings on.

the discuss list remains quite valuable to me, and is high signal and low noise.

teleconferences and synchronous communication involves a lot of effort for me, since I'm already saturated with meetings during working hours, and have a family.

github is wonderful for asynchronous tracking of many disparate issues (flexible notification system, connects with my inbox well, links between issues, etc.) and is my preferred engagement tool of choice.

a forum that is well integrated with email (I am trying out discourse) could be a valuable way to spur more fine grained community interaction on specific topics.

HTH!

**Comment by wrightaprilm** [comment]
Hi @serahrono,

You've caught me on Mardi Gras break, so wall of text incoming.

**General:** 

I don't feel comfortable commenting on the discuss listserve (not willing to say more in this venue, but OK commenting privately). In a rapidly-scaling and diverse community, things like that will occur. I read the Twitter feed (where I saw this), and use the slack channel. I hope the organization will support marginalized members in managing communications by not confining mission critical info to one particular channel. I generally feel that I'm getting the information I need. But I can see where there are gaps, such as the spreadsheet of teaching opportunities only going to the listserve. If someone sent a lesson-breaking bug in the lesson I maintain to discuss, it might not get handled. My email marks as read and filters directly - I simply wouldn't see it unless pinged directly via GitHub, Slack or Twitter. 

**Maintainer Specific:**

Maintainer is a little bit of a problematic role in all communities, eh? I think how maintainers communicate with the community could use a little work, and I'm not sure how to fix it. When I get a pull request or issue with a radical change, I usually solicit feedback from the community. The problem is that whenever I do that, I often spend more time looking for people to even comment than I would have just merging or giving the feedback myself. It's like being a journal editor! This gets to the heart of the role itself - is maintainer maintenance or is maintainer creative? It's a little bit of both, but I think improving communication with the broader community could help tip towards "maintenance."

It would be great if there was some central board of issues, or if we could get pinged befor instructor trainings so we could forward some of the more pressing and novice friendly issues to the trainer trainers. Or if there were guidelines for how to get a request for help tweeted on our behalf. Some of this will, presumably, improve with the update lesson release cycle and better adverts of Bug BBQs. But I think the ongoing process of Feedback --> Updates --> Improvement could be made more collaborative with better communication.

**Comment by dvanic** [comment]
Hi! 

As a "research engineer" (for lack of a better internationally-translatable term, my official title is "Data Analytics Trainer and Data Scientist"), I'd like to comment on the suggestions @ctb put forth, and make a few of my own suggestions:

> periodic e-mails are great for general notifications!

Agreed! They can also be good for starting discussions, although I have often found that I wish some of the more interesting conversations were a blog post where we could comment instead (we all get so many emails everyday, and most are not high-value, so having cool Carpentries stuff there is not necessarily a good thing). Perhaps allow more people to post questions to the Carpentries blog (like they're currently doing with discuss), and then flick an email to discuss asking for comments on the blog?

>twitter 

I'm of two minds about twitter. On one hand, it's quite good at building community and connecting people, but on the other, it promotes shallow work and the notification itch, which can be quite stressful (at least for me) - so while I've used it sporadically over the last few years, it's not something that has really "stuck", and I know lots of academics and non-academics who fundamentally refuse to engage with it as a platform. So I'd like to beg that the Carpentries don't choose this as an engagement format.

> blogging remain my main officially supported mechanisms for keeping up with the Carpentries goings on.

I love blogging, and the idea of cross-posting to individual blogs + the Carpentries one (or just the Carpentries, if someone doesn't have their own).

I also think, though, that it's critical to differentiate between (1) informative posts, such as discussions of where to find materials and how to teach etc vs (2) update/scheduling/operations posts, which may be better communicated via email, actually, since (for better or worse) that's the medium we all check, and use for that purpose anyway.

> the discuss list remains quite valuable to me, and is high signal and low noise.
As mentioned above, I think a lot of what's on discuss would be more pleasant to engage with if it weren't in email, but was instead on the blog?....

> teleconferences and synchronous communication involves a lot of effort for me, since I'm already saturated with meetings during working hours, and have a family.

Agreed - but they're also the only form of face-to-face communication many of us have with the broader community (especially as we're in faraway Australia), and we're really trying to get some more "down under" friendly time zone meetings to happen. My family has gotten used to the fact that mommy often has 5 am zoom meetings (but I can't do 1 am, for the life of me, which is most of what East Coast US morning ends up being). I think giving lots of notification, via email (optional opt-in, if it doesn't work for everyone), at least a week in advance, would help me plan for this better.

> github is wonderful for asynchronous tracking of many disparate issues (flexible notification system, connects with my inbox well, links between issues, etc.) and is my preferred engagement tool of choice.

Agreed! As an alternative, lower profile/lower stakes platform I think it's great. Could be an alternative to blogging for the "smaller" issues. 

> a forum that is well integrated with email (I am trying out discourse) could be a valuable way to spur more fine grained community interaction on specific topics.

Haven't tried this, so can't comment

> Slack

Another tool that the community uses, with which, again, only a small portion engage with.

*** 

Overall, I think, communication approaches for me fall into 2 broad categories:

- Notification of upcoming events, logistical updates etc (i.e. things that are irrelevant if communicated in 12 months). My platform of choice for these would be email (as opposed to twitter or slack), purely because that's already where all of the work/life/non-Carpentries communication about such things lives.

- Communication regarding "deeper" things (i.e. relevant today or in 12 months or 12 months ago). At the moment, a lot of these are coming in via discuss, some are on twitter, while others yet are on the blog, github and some are even on slack! I *hate* the fact that I have to check all of these venues, and this frustrates/stresses me out in equal measure; I always feel like I'm behind, and am missing things that would be very useful for my teaching (not just Carpentries). I'd love for all of these to be in one place, probably the blog, as it would allow me to turn on notifications once, then find out about new posts, engage with them when I have the time (i.e somewhat asynchronously), and bookmark them for future reference/followup when I'd like to remind myself about what was said 4 months from now. 

- A "lower-stakes" complement to the blog could be a separate "discuss" repo, which once again I could turn on notifications for, and engage with when I have the time.

***

**TLDR: email for time-sensitive updates, blog + github for discussions**

**Comment by NonglongSu** [comment]
> It has been two years since this discussion started, and several changes have happened in that time, key among them being that [Library Carpentry became an official Lesson Program of The Carpentries in November 2018](https://carpentries.org/blog/2018/11/welcoming-library-carpentry/).
> 
> We are currently working on updating The Carpentries communications strategy, and as such, we would like to revisit this exchange and are keen to hear from you on a number of things:
> 
>     * what’s currently working, and what’s not, about
>       
>       * the ways we communicate with you as members our community, and
>       * the ways we enable you to interact with others in the community?
> 
>     * what aspects of our communiqué do you still have questions about or need clarity on, be it information in a resource we've shared with you, or a page on our website, etc?
> 
>     * do you have any suggestions on how we can
>       
>       * communicate more effectively with you going forwards and/or
>       * facilitate better interactions between you and other members of The Carpentries community?
> 
> 
> Your feedback on any of the questions above really matters to us, and we will inform our comms strategy drafting process and implementation later on.
> 
> [Submissions via email](mailto:serah@carpentries.org) are also welcome.

Hi, Serah, 
      I understand the importance of communication in different local places, and translations seems to be a core problem,. If we can make a good translation effort which means we already build a stable bridge for passing knowledge and information. 
     As an international student from China, I understand that platform access would be not easy in my country. From my perspective , I can think of several main reasons: 1. There is no dependable access to google. Google is still limited in most of the cities in China which would limit the access to updated information from local students. 2. I am sure that we are not lack of good programmers or teachers in China, but how to attract their attention and gather them so we could build a local workshop and  raise donations. (I suggest we could start from some universities that own the collaborative programs with China. (For instance, ASU has the 3+2, 4+1 programs with Sichuan university, China, so it is much easier to let the international student studying here to go back to tell their schoolfellows what is software carpentry. 

Currently, these are two main issues that I can think of.

**Comment by sstevens2** [comment]
I really love Carpentry Clippings.  Even if I missed things in discuss or the blog or github, I basically depend on finding the important stuff in the Clippings.

It does seem like there are a lot of different channels of communication, discuss/email lists, slack, twitter, blog.  While I use and like most of these forms of communication, it can be unclear when to use which medium.  Maybe if we had some guidelines about when to use each it might be easier.

Around slack, I love slack so that is great but I have a pet peeve, people responding not in thread.  I get so many notifications when a discussion is going on and people aren't using threads.  So I guess I'd like this to be part of the guidelines for slack.  Others may have different opinions on this though.

Following up on what @wrightaprilm said about being a maintainer.  I'm not a maintainer but it would be nice if there was an easier way to solicit community feedback.  I think there are community members who would be interested in giving opinions on suggested changes or helping with issues but they don't always know anything is happening.  Maybe this is getting people to 'watch' repos more and have suggestions for how they can get involved with issues, maybe it is a list of people who volunteer to look at issues for a specific lesson when a maintainer needs feedback?  Not sure best practices there.  

Thanks for the opportunity to give communications feedback!

**Comment by wrightaprilm** [comment]
>  I think there are community members who would be interested in giving opinions on suggested changes or helping with issues but they don't always know anything is happening. 

This is a really fantastic point, @sstevens2. We actually do have a better track record in the py-ecology repo at contributorship than other open source projects I'm involved with. A quick eyeball of the number of forks (~175) and the number of unique pull requesters (~35) bears that out. There's a lot of enthusiasm there.

We do have ~80 folks who have a watch on the repo, but I've never seen any of them (besides paid Carpentries staff) comment without a direct tag in. And I think that's natural - lessons tend to have a feast-or-famine cycle of interest. Often, a bunch of issues get opened while someone is going through getting ready to teach. In the issue flurry, people put the repo on mute, and never unmute. 

The place I have the most luck recruiting help is Twitter, but I feel uncomfortable using my personal account to request help for organizational issues too often. Something to consider might be a Google form where people doing work for the community could fill in:

- What is the problem (+ link to PR/issue)
- Requested actions/expertise
- Other pertinent info

To get the info tweeted or included in Carpentries Clippings? 

And @sstevens2 - I know you're not a maintainer, but in the April 17 maintainers meeting (connection details [here](https://pad.carpentries.org/maintainers)), I'll be talking about the various pressures in soliciting, digesting, and balancing feedback from the community. It'll be centered on the maintainer role, but I'm sure you'd have some great ideas, and you'd be very welcome to participate.

**Comment by ShailyGandhi** [comment]
Hi, Serah,
I understand the importance of communication in different local places, and Like most of the carpentries, communication happens Twitter, slack channel and GitHub, Zoom Calls, emails with the spreadsheet of teaching opportunities. But I can see where there are gaps, like looking at the scenario in India people are not using these modes of communication more frequently. Most used ones are Facebook, Instagram, Linkedin, whats app, Emails are more often used for communication and telephonic follow up is also required at times. 

There is a handful of few people in India who are certified instructors who are spatially located in different areas and a big communication gap. I would like to make it possible to use the local platforms to spread a word about the upcoming events and happenings so that it reached the mass.

**Comment by raynamharris** [comment]
Thanks @serahrono for reopening the conversations. I have a rant and a question/concerns. 

1. The rant: I know it's important to send information via all the channels (email, blog, tweet, slack) to reach the broadest audience. I don't mind seeing the same content in multiple places. I do mind, however, when there are multiple reminders about the same thing in the same week via all platforms. Maybe there could be efforts to improve the spacing of reminders.

2. The question/concern: For me, it's unclear when community-driven ideas need to be communicated to staff or leadership for approval and/or simply as an FYI. It's also unclear where these should be directed. Often I ask a specific person I trust or who engages with that topic, but maybe these questions should go to a general inbox...

**Comment by serahrono** [comment]
thanks @ctb, @wrightaprilm, @dvanic, @NonglongSu, @sstevens2, @ShailyGandhi, @raynamharris and @rgaiacs for taking time to provide feedback and ask questions about our current comms. We've read each one, and are actively working to incorporate your input in our comms implementation plan for the coming days. Over the next month, you can expect updates about any specific concerns/questions you raised to be shared in this thread, and you will be tagged. Also feel free to check in with me at any point in the meantime.
